posthog/api/test/test_capture.py

```diff
@@ -2366,3 +2366,176 @@ def test_capture_replay_does_not_write_to_bucket(self, _name: str, sample_rate:
 
             with pytest.raises(ObjectStorageError):
                 object_storage.read("token-another-team-token-session_id-abcdefgh.json", bucket=TEST_SAMPLES_BUCKET)
+
+    @patch("posthog.kafka_client.client._KafkaProducer.produce")
+    def test_capture_csp_violation(self, kafka_produce):
+        csp_report = {
+            "csp-report": {
+                "document-uri": "https://example.com/foo/bar",
+                "referrer": "https://www.google.com/",
+                "violated-directive": "default-src self",
+                "effective-directive": "img-src",
+                "original-policy": "default-src 'self'; img-src 'self' https://img.example.com",
+                "disposition": "enforce",
+                "blocked-uri": "https://evil.com/malicious-image.png",
+                "line-number": 10,
+                "source-file": "https://example.com/foo/bar.html",
+                "status-code": 0,
+                "script-sample": "",
+            }
+        }
+
+        response = self.client.post(
+            f"/csp/?token={self.team.api_token}",
+            data=json.dumps(csp_report),
+            content_type="application/csp-report",
+        )
+
+        self.assertEqual(status.HTTP_204_NO_CONTENT, response.status_code)
+        self.assertEqual(kafka_produce.call_count, 1)
+
+        kafka_produce_call = kafka_produce.call_args_list[0].kwargs
+
+        # Verify data
+        event_data = json.loads(kafka_produce_call["data"]["data"])
+
+        self.assertEqual(event_data["event"], "$csp_violation")
```

Comment:
I won't say we should change it as a blocking step here

i think we should always prefer the `assert x == y` form
we have a plugin that makes the failure output more clear

Reply:
I will do that